File: app/src/main/java/com/yuta/__old/projects/adapter/ProjectTeamSearchAdapter.java
package com.yuta.__old.projects.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import androidx.annotation.NonNull;
import com.yuta.__old.R;
import com.yuta.app.domain.model.entity.Team;
import com.yuta.common.ui.BaseAdapter;
import com.yuta.common.ui.AppDialog;
import com.yuta.__old.projects.dialog.CreateProjectDialog;

import java.util.List;

import static android.view.View.GONE;
import static android.view.View.VISIBLE;

public class ProjectTeamSearchAdapter extends BaseAdapter<Team, BaseAdapter.ViewHolder<Team>> {
    private final AppDialog dialog;
    private final ProjectTeamSearchAdapter addedTeamSearchAdapter;

    public ProjectTeamSearchAdapter(AppDialog dialog, List<Team> items, ProjectTeamSearchAdapter teamSearchAdapter) {
        super(dialog.getContext(), items);
        this.dialog = dialog;
        this.addedTeamSearchAdapter = teamSearchAdapter;
    }

    @NonNull
    @Override
    public TeamViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_team_search_in_project, parent, false);
        return new TeamViewHolder(view);
    }

    public class TeamViewHolder extends BaseAdapter.ViewHolder<Team> {
        private final TextView name;
        private final Button buttonAdd;
        private final Button buttonRemove;

        public TeamViewHolder(@NonNull View itemView) {
            super(itemView);
            this.name = itemView.findViewById(R.id.name);
            this.buttonAdd = itemView.findViewById(R.id.add_team);
            this.buttonRemove = itemView.findViewById(R.id.remove_team);
        }

        @Override
        public void bind(Team team) {
            name.setText(team.getName());

            if (addedTeamSearchAdapter != null) {
                buttonRemove.setVisibility(GONE);
                buttonAdd.setVisibility(VISIBLE);

                buttonAdd.setOnClickListener(v -> {
                    List<Team> addedItems = addedTeamSearchAdapter.getItems();
                    if (!addedItems.isEmpty()) {
                        addedTeamSearchAdapter.removeItem(addedItems.get(0));
                    }

                    removeItem(team);
                    addedTeamSearchAdapter.insertItem(team);
                    ((CreateProjectDialog) dialog).updateSubmitButtonState();
                    ((CreateProjectDialog) dialog).updateAddedTextVisibility();
                });
            } else {
                buttonAdd.setVisibility(GONE);
                buttonRemove.setVisibility(VISIBLE);

                buttonRemove.setOnClickListener(v -> {
                    removeItem(team);
                    ((CreateProjectDialog) dialog).updateSubmitButtonState();
                    ((CreateProjectDialog) dialog).updateAddedTextVisibility();
                });
            }
        }
    }
}

File: app/src/main/java/com/yuta/projects/ui/adapter/ProjectTeamSearchAdapter.kt
package com.yuta.projects.ui.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.View.GONE
import android.view.View.VISIBLE
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import com.yuta.app.R
import com.yuta.common.ui.BaseAdapter
import com.yuta.domain.model.Team

class ProjectTeamSearchAdapter(
    items: MutableList<Team>,
    private val addedTeamSearchAdapter: ProjectTeamSearchAdapter? = null,
    private val onUpdateCallback: () -> Unit
) : BaseAdapter<Team, BaseAdapter.ViewHolder<Team>>(items) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TeamViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_team_search_in_project, parent, false)
        return TeamViewHolder(view)
    }

    inner class TeamViewHolder(itemView: View) : ViewHolder<Team>(itemView) {
        private val name: TextView = itemView.findViewById(R.id.name)
        private val buttonAdd: Button = itemView.findViewById(R.id.add_team)
        private val buttonRemove: Button = itemView.findViewById(R.id.remove_team)

        override fun bind(team: Team) {
            name.text = team.name

            addedTeamSearchAdapter?.let {
                buttonRemove.visibility = GONE
                buttonAdd.visibility = VISIBLE

                buttonAdd.setOnClickListener {
                    if (addedTeamSearchAdapter.items.isNotEmpty()) {
                        addedTeamSearchAdapter.removeItem(addedTeamSearchAdapter.items[0])
                    }

                    removeItem(team)
                    addedTeamSearchAdapter.insertItem(team)
                    onUpdateCallback()
                }
            } ?: run {
                buttonAdd.visibility = GONE
                buttonRemove.visibility = VISIBLE

                buttonRemove.setOnClickListener {
                    removeItem(team)
                    onUpdateCallback()
                }
            }
        }
    }
}
